### PR TITLE
cloud: add storage support

### PR DESCRIPTION
**Pull request by octonawish-akcodes**
Cloud storage deployment support.

### CODE REVIEW

**Comment by octonawish-akcodes** [comment]
> Uh, there's still a bunch of things to fix from the last review. Please don't request another review until you addressed or argued the previous review's comments.

I mistakenly clicked on it, sorry ://

File: cloud
Comment (spbnick) [inline cloud:18]: Please do not change this `export` to `declare -r`. We *need* this exported.

File: cloud
Comment (spbnick) [inline cloud:1253]: Um, you are still not setting the original `TMPDIR` here, as I suggested.

**Review by spbnick — CHANGES_REQUESTED** [review]

File: cloud
Comment (tales-aparecida) [inline cloud:1248]:
```suggestion
# Args: project bucket
```

File: cloud
Comment (tales-aparecida) [inline cloud:1257]:
```suggestion
# Args: bucket
```

File: cloud
Comment (octonawish-akcodes) [reply cloud:1253]: When I was testing it with those code changes I didn't got those errors, but on further testing I got those errors again. Updated the code and tested to all works fine.

File: cloud
Comment (octonawish-akcodes) [reply cloud:18]: Okay

File: cloud
Comment (spbnick) [reply cloud:1253]: So, will you add setting the `TMPDIR`?

Comment (octonawish-akcodes) [reply cloud]: Yes

File: cloud
Comment (spbnick) [inline cloud:1252]: Please wrap this line to fit under the 78 column limit, as I requested earlier.

**Review by spbnick — CHANGES_REQUESTED** [review]

File: cloud
Comment (octonawish-akcodes) [reply cloud]: Done

File: cloud
Comment (spbnick) [inline cloud:1253]: Please add one more indent level to this line, so it's clearer it's continuing from the previous one, as is done for the rest of the file. It's easy to miss just the backslash otherwise.

File: cloud
Comment (spbnick) [inline cloud:16]: Please add a comment above this line, like I had in my example for this variable.

Comment (octonawish-akcodes) [reply cloud:17]: done

File: cloud
Comment (octonawish-akcodes) [reply cloud]: done

File: cloud
Comment (spbnick) [inline cloud:1255]: Don't edit this now, but it's good to wrap the option name *together* with the option value, because then the command is easier to read.

File: cloud
Comment (spbnick) [inline cloud:1263]: Why do you use the `-m` option here?

**Review by spbnick — COMMENTED** [review]
Thank you! Looks good to me now, but I have one question inline (and one comment for later notice)

File: cloud
Comment (octonawish-akcodes) [reply cloud:1263]: Without it the object of the bucket will get delete one by one but `-m` parallel deletions can be done which can make process faster of deletion, [Supported Link](https://cloud.google.com/storage/docs/gsutil/addlhelp/GlobalCommandLineOptions)

File: cloud
Comment (octonawish-akcodes) [reply cloud]: Updated

**Review by spbnick — APPROVED** [review]
LGTM! Thank you, @octonawish-akcodes. Let's wait for CI results before merging.

File: cloud
Comment (octonawish-akcodes) [reply cloud:1263]: okay

**Pull request merged**
